### DISCUSSION TITLE
increase retry timeout

### DISCUSSION THREAD

**Discussion by test2a** [rustdesk/rustdesk#3825] [Ideas]
i am used to " remote desktop", or "RDP" whereby it does a  reconnection request (test) every 5 seconds  for a  total of 20 times so around 1.6 minutes.

Our current timeout is too small in terms of bad connections because i have occasional drops and sometimes it takes a few seconds, sometimes 20-40. maybe more. 
i have auto set password to the remote machine so i should have a "timeout configure" option so if i set say "1 minute", the app will retry for 1 minute before failing.

this is helpful for a lot of reasons and people who get spotty internet or connection drops somehow, this should be able to help....
https://social.technet.microsoft.com/Forums/windows/en-US/4a60f36e-143f-4e4b-a565-aecc68e0238a/rdp-reconnect-attempt-intervals

we could have the same "feature" for retesting the connection and an option to configure longer attemps or intervals (or both) would be also great

**Comment by Kylejustknows** [comment]
Yea, I manage 400+ active computers via Rustdesk.

Two headache problems:

1, I can't set a default password....typing the same password hundreds of times per day is a headache.
2, The default reconn timeout is an ill design, because:

- After timeout, the tab hang, if click into it, it says connection error, then I must click to close the tab....**What's the purpose of timeout?** it suppose to save us time and effort. If it can't save time and effort, why even making the timeout dialog? 
- The timeout is shorter than most PC reboot time. Meaning reboot -> retry -> retry -> timeout error -> I click to close tab -> re-conn...... for a single PC reboot. Now in order to find an "operation consistency", I kill it whenever I see the re-try dialog, and feel much better.
- Sometimes there is re-try dialog, sometimes it is just saying connection closed.  Unpredictable.

The re-try takes nearly 0 resource locally and remotely.  Why don't we do it more often, and longer period?

E.g.
If Rustdesk auto re-conn every 1 second, notice hide in corner, and keep re-try no matter what, until user close it....This will give Rustdesk much smoother end-user experience: remote control just like an local physical monitor (instead of bunch of popups, tab closing, reconnect etc.)
